modules/gapi/test/own/rmat_view_tests.cpp

```diff
@@ -0,0 +1,147 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#include "../test_precomp.hpp"
+#include <opencv2/gapi/rmat.hpp>
+
+#include <opencv2/gapi/util/compiler_hints.hpp>
+#include "../src/backends/common/gbackend.hpp"
+
+namespace opencv_test
+{
+using cv::GMatDesc;
+using View = cv::RMat::View;
+using cv::Mat;
+using namespace ::testing;
+
+static void expect_eq_desc(const View& view, const GMatDesc& desc) {
+    EXPECT_EQ(view.size(), desc.size);
+    EXPECT_EQ(view.dims(), desc.dims);
+    EXPECT_EQ(view.cols(), desc.size.width);
+    EXPECT_EQ(view.rows(), desc.size.height);
+    EXPECT_EQ(view.type(), CV_MAKE_TYPE(desc.depth,desc.chan));
+    EXPECT_EQ(view.depth(), desc.depth);
+    EXPECT_EQ(view.chan(), desc.chan);
+}
+
+TEST(RMatView, TestDefaultConstruction) {
+    View view;
+    GMatDesc desc{};
+    expect_eq_desc(view, desc);
+    EXPECT_EQ(view.ptr(), nullptr);
+    EXPECT_EQ(view.step(), 0u);
+}
+
+struct RMatViewTest : public TestWithParam<int>{};
+TEST_P(RMatViewTest, ConstructionFromMat) {
+    auto type = GetParam();
+    Mat mat(8,8,type);
+    const auto desc = cv::descr_of(mat);
+    View view(cv::descr_of(mat), mat.ptr(), mat.step1());
+    expect_eq_desc(view, desc);
+    EXPECT_EQ(view.ptr(), mat.ptr());
+    EXPECT_EQ(view.step(), mat.step1());
+}
+
+TEST(RMatView, TestConstructionFromMatND) {
+    std::vector<int> dims(4, 8);
+    Mat mat(dims, CV_8UC1);
+    const auto desc = cv::descr_of(mat);
+    View view(cv::descr_of(mat), mat.ptr());
+    expect_eq_desc(view, desc);
+    EXPECT_EQ(view.ptr(), mat.ptr());
+}
+
+TEST_P(RMatViewTest, DefaultStep) {
+    auto type = GetParam();
+    GMatDesc desc;
+    desc.chan = CV_MAT_CN(type);
+    desc.depth = CV_MAT_DEPTH(type);
+    desc.size = {8,8};
+    std::vector<unsigned char> data(desc.size.width*desc.size.height*CV_ELEM_SIZE(type));
+    View view(desc, data.data());
+    EXPECT_EQ(view.step(), static_cast<size_t>(desc.size.width)*CV_ELEM_SIZE(type));
+}
+
+static Mat wrapViewByMat(View& view) {
+    return Mat(view.size(), view.type(), view.ptr(), view.step());
+}
+
+TEST_P(RMatViewTest, StepIn) {
```

Comment:
not very descriptive name (the next test as well). What was meant here ?

Reply:
Renamed